### DISCUSSION TITLE
Complete beginner, looking for advice on learning techniques.

### DISCUSSION THREAD

**Discussion by strencis** [codewars/codewars.com#2603] [Q&A] [answered]
Hello everybody!

I am through 25% of an online course on Udemy, "From zero to expert". I find it to be really great and comprehensive, but obviously it cannot include everything that there is possibly to know. Also, I have some minor experience with fundamentals before.  After going through the Fundamentals Part 1 and 2 of this course I decided that it is finally time to do some actual coding myself and went on to start with Codewars challenges. BUT. I find it to be quite complicated to solve even the simplest of the challenges, mostly due to Methods that I don`t even know exist. Finished some of the challenges after extensive googling and comparing to finished code(sometime copying it) and obviously went to see the solutions on the ones that I was getting really frustrated on. Most of the time I am really close on the logic, like where to use loops, if/else statements, switch etc. but the MAIN problem is I am unaware of most of the methods and the possibilities that JS can offer, sometimes also syntax (which CW does not show, like a browser console does). Feels like I am hitting a wall just because of that and it`s becoming increasingly frustrating, because I don`t see how should I know some of that stuff. Is there any advice out there on how should I approach this - documentation, books, resources that I should maybe read and get to know before this? It does not feel like I am progressing if I keep looking for solutions instead of coming up with the solutions myself, weather it`s on google or within the CW solutions. Any advice would be much appreciated. Don`t really have a coding buddy to struggle with together - everybody I know is already a developer and does not have that much time to spend on the very fundaments with me. The struggle is real, but I am 100% committed. Started going through the beginning of the course again today in case I missed some stuff, but that seems pointless.

Sincerely,
Mik

**Comment by hobovsky** [comment, marked as answer]
For me' it's always been that the best source of knowledge are people smarter and more experienced than me. You said that you have no coding buddy, but there's many such buddies ready to help in various communities, including Codewars. You can join Codewars Discord and seek for help on `#beginners` or `#javascript`, and people there are quite willing to offer some hints, guidance, and advice.

However, it does not mean that you will be given complete solutions. Googling, reading, and searching for solutions is a skill which is invaluable while solving problems. At the beginning, everything is difficult and becomes easier with experience, and even when you ask others for help, you still need to be ready to search through the Internet, videos, tutorials, and courses to find this exact piece which will make your solution work. Someone might tell you _"You need to make your solution faster, `array.contains` is too slow"_, or _"There's a function in `regex` module which can solve this problem for you"_, etc. and then you need to take things from there and google for _"contains on an array too slow what to use instead"_, or ask follow-up questions.

At the beginning, it's always difficult. Codewars problems also have quite high entry threshold for absolute, total beginners. But they can be solved by a determined user asking sufficient amount of questions.

Good luck!

**Comment by placoderm** [comment]
Congratulations on the progress you have made so far. 

I think that getting comfortable with frustration is the best skill to develop. We are always going to be pushing up against the things we don't know.

BTW, a tip about markdown formatting. Looks like you are using the back tick ( ` ) instead of an apostrophe in your message. That's why some of the text has a box around it. The back tick should never be used in English writing. Stick to the regular apostrophe( ' ) You can also use three back ticks before and after a code block to format it as such. Just google MarkDown formatting. 

Welcome!

**Comment by NickVasenkov** [comment]
One may assume that a well-developed programming language has shortcuts for most of the common tasks. One just needs to break down the problem into the common tasks and google those that are not learned by them yet.

**Reply by mousewanderer** [reply]
Yeah I know right

**Comment by napalmtango** [comment]
The process you describe is par for the course. Try not to get too frustrated and do understand that having to search Google, many times in vain, is not a bug but a feature. To restate: The skill of finding solutions online is far more important than any actual technical skills you will learn. Again, try not to let the natural reaction of frustration get to you. This is normal. And is, paradoxically, a sign that you both care and are making progress. Furthermore, you will be conducting this process for as long as you are coding. Keep at it and good luck to you!

**Reply by jeromejohnmba** [reply]
Nailed it.

**Comment by AbdoAboderais** [comment]
i am in the same situation like you, but i found out sometimes you need to stick with the code problem and try to solve it, instead of copy the solution from internet,  mostly you will your way at least the logic to solve it.

**Comment by jeromejohnmba** [comment]
Courses will teach you next to nothing. I took that same course. When I finished I knew almost nothing still. You have to solve a ton of problems and start small. Use pseudocode a lot. That helps.

**Comment by jeromejohnmba** [comment]
I noticed also that I would come to a solution later as an epiphany.

**Comment by webForDave** [comment]
What I do is, I sit for a while and think about the challenge. 
Think about what solutions would work, think about what concepts are required and all that.

If I'm stuck and can't move on, I go on Google and search for something related to the question and not the question in particular. I I'm truly stuck, I just go on checking the solution. The point here is STRUGGLE.
